packages/core/src/codecs/octetstream-codec.ts

```diff
@@ -17,6 +17,11 @@ import { ContentCodec } from "../content-serdes";
 import { DataSchema, DataSchemaValue } from "wot-typescript-definitions";
 import { getFloat16, setFloat16 } from "@petamoriken/float16";
 
+import debug from "debug";
+
+const logDebug = debug("core/octetstream-codec");
+const logWarn = logDebug;
+
```

Comment:
Since I am not familiar with the debug library I was wondering how "we" or "users" of node-wot can configure to log only warnings/errors besides *all* debug statements?

Moreover, wouldn't it make sense to add a prefix to make clear which kind of log statement it is? Like the following?
const logDebug = debug("[debug] core/octetstream-codec");
const logWarn = debug("[warn] core/octetstream-codec");

Reply:
Hmm, maybe we should actually consider using [debug-level](https://www.npmjs.com/package/debug-level) instead then, as it would give us both capabilities and more.

Reply:
Makes sense to me.
@relu91?

Reply:
Some concerns about `debug-level`:
- It is not that popular
- It seems to have just one maintainer
- It has quite a few dependencies for a logging library
- It depends directly on debug but it implements its own logic (I actually don't understand why the dev choose this path)
- I think we can use namespaces to achieve "almost" the same behaviour example below

```js
const debug = debug("core:octetstream-codec:debug");
const info = debug("core:octetstream-codec:info");
```

Notice that the *well-known* separator is `:` as document [here](https://www.npmjs.com/package/debug#Conventions). Enable or disable level nails down to correctly set up the environment variable:
```bash
DEBUG=core:*:info # any info core message
DEBUG=core:* # any core message
```
We sort of lose di ability to say I want ALL the messages from WARN to INFO but I think is not really so problematic, because you can still do this:
```bash
DEBUG=core:*:info, core:*:warn # any core info and warn messages
```

Therefore I would still go for the `debug` library, what do you think?

Reply:
This path is fine by me also. We just need to carefully document this or point to a proper documentation so that people know how to configure debugging properly.

Reply:
Thank you, @relu91, explanation of the correct usage of the debug module! I now applied the separator/namespace syntax and also did some refactoring, locating the functions for creating the loggers again in a `loggers.ts` file.

The functions defined here allow you to either create a logging function for one specific log level (using, for example, `createDebugLogger` or `createErrorLogger`) or create multiple loggers at once, like so: 

```ts
const { debug, warn } = createLoggers("core", "servient");
```

When creating loggers, the first argument is always the package name and the second one is the namespace for the current file, the two of which get separated by a colon when creating the `Debugger` object as you described. The respective log level (e.g., `debug`) is always appended at the end, so that the full namespace of the `debug` logger above would become `core:servient:debug`, while the full `warn` namespace would be `core:servient:warn`.

I generally have a good feeling about this new solution, and I am looking forward to your feedback :)